js/src/components/editor/DiffEditor.tsx

```diff
@@ -0,0 +1,152 @@
+"use client";
+
+import { PostgreSQL, sql } from "@codemirror/lang-sql";
+import { yaml } from "@codemirror/lang-yaml";
+import { MergeView, unifiedMergeView } from "@codemirror/merge";
+import { EditorState, Extension, Text } from "@codemirror/state";
+import { EditorView } from "@codemirror/view";
+import { githubDark, githubLight } from "@uiw/codemirror-theme-github";
+import { useEffect, useMemo, useRef } from "react";
+
+export type DiffEditorLanguage = "sql" | "yaml";
+export type DiffEditorTheme = "light" | "dark";
+
+export interface DiffEditorProps {
+  original: string;
+  modified: string;
+  language?: DiffEditorLanguage;
+  readOnly?: boolean;
+  lineNumbers?: boolean;
+  sideBySide?: boolean;
+  height?: string;
+  className?: string;
+  /** Theme: "light" for githubLight, "dark" for githubDark */
+  theme?: DiffEditorTheme;
+  /** Called when modified content changes (only if not readOnly) */
+  onModifiedChange?: (value: string) => void;
+}
+
+const getLanguageExtension = (language: DiffEditorLanguage) => {
+  switch (language) {
+    case "sql":
+      return sql({ dialect: PostgreSQL });
+    case "yaml":
+      return yaml();
+    default:
+      return sql({ dialect: PostgreSQL });
+  }
+};
+
+const getThemeExtension = (theme: DiffEditorTheme): Extension => {
+  return theme === "dark" ? githubDark : githubLight;
+};
+
+export function DiffEditor({
+  original,
+  modified,
+  language = "sql",
+  readOnly = true,
+  lineNumbers = true,
+  sideBySide = true,
+  height = "100%",
+  className = "",
+  theme = "light",
+  onModifiedChange,
+}: DiffEditorProps) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const viewRef = useRef<MergeView | EditorView | null>(null);
+
+  const extensions = useMemo(() => {
+    return [getLanguageExtension(language), getThemeExtension(theme)];
+  }, [language, theme]);
+
+  useEffect(() => {
+    if (!containerRef.current) return;
+
+    // Clean up previous instance
+    if (viewRef.current) {
+      viewRef.current.destroy();
+    }
+
+    if (sideBySide) {
+      // Side-by-side mode using MergeView
+      const mergeView = new MergeView({
```

Comment:
uiwjs also provides the CodeMirrorMerge React component.
https://uiwjs.github.io/react-codemirror/#/merge/document
We don't need to implement the merge view by ourselves.